sdcm/cluster_aws.py

```diff
@@ -185,14 +186,36 @@ def _create_instances(self, count, ec2_user_data='', dc_idx=0):
                        'SubnetId': self._ec2_subnet_id[dc_idx],
                        'AssociatePublicIpAddress': True,
                        'Groups': self._ec2_security_group_ids[dc_idx]}]
-
-        if self.instance_provision == INSTANCE_PROVISION_ON_DEMAND:
+        if self.instance_provision == 'mixed':
+            instances = self._create_mixed_instances(count, interfaces, ec2_user_data, dc_idx)
+        elif self.instance_provision == INSTANCE_PROVISION_ON_DEMAND:
             instances = self._create_on_demand_instances(count, interfaces, ec2_user_data, dc_idx)
         else:
             instances = self._create_spot_instances(count, interfaces, ec2_user_data, dc_idx)
 
         return instances
 
+    def _create_mixed_instances(self, count, interfaces, ec2_user_data, dc_idx):
+        instances = []
+        if isinstance(self, ScyllaAWSCluster) or isinstance(self, CassandraAWSCluster):
+            if count > 2:
+                count_on_demand = 2
+            else:
+                count_on_demand = 1
+            self.instance_provision = INSTANCE_PROVISION_ON_DEMAND
+            instances.extend(self._create_on_demand_instances(count_on_demand, interfaces, ec2_user_data, dc_idx))
+            self.instance_provision = INSTANCE_PROVISION_SPOT_LOW_PRICE
+            instances.extend(self._create_spot_instances(count - count_on_demand, interfaces, ec2_user_data, dc_idx))
+        elif isinstance(self, LoaderSetAWS):
+            self.instance_provision = INSTANCE_PROVISION_SPOT_LOW_PRICE
+            instances = self._create_spot_instances(count, interfaces, ec2_user_data, dc_idx)
+        elif isinstance(self, MonitorSetAWS):
+            self.instance_provision = INSTANCE_PROVISION_ON_DEMAND
+            instances.extend(self._create_spot_instances(count, interfaces, ec2_user_data, dc_idx))
```

Comment:
Quoted from the commit log : `- monitoring nodes with instance type - ON_DEMAND always`

Here the instance_provision is INSTANCE_PROVISION_ON_DEMAND, but it called `_create_spot_instances()`. why not `_create_on_demand_instances()`?

Reply:
You right. it is mistake.

Reply:
change to create_on_demand